fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayTabLayouts.kt

```diff
@@ -34,85 +34,86 @@ import org.mozilla.fenix.theme.FirefoxTheme
  * Top-level UI for displaying a list of tabs.
  *
  * @param tabs The list of [TabSessionState] to display.
+ * @param displayTabsInGrid Whether the tabs should be displayed in a grid.
  * @param onTabClose Invoked when the user clicks to close a tab.
  * @param onTabMediaClick Invoked when the user interacts with a tab's media controls.
  * @param onTabClick Invoked when the user clicks on a tab.
  * @param onTabLongClick Invoked when the user long clicks a tab.
+ * @param header Optional layout to display before [tabs].
  */
+@Suppress("LongParameterList")
 @Composable
 fun TabList(
     tabs: List<TabSessionState>,
+    displayTabsInGrid: Boolean,
     onTabClose: (TabSessionState) -> Unit,
     onTabMediaClick: (TabSessionState) -> Unit,
     onTabClick: (TabSessionState) -> Unit,
     onTabLongClick: (TabSessionState) -> Unit,
+    header: (@Composable () -> Unit)? = null,
 ) {
     val tabListBottomPadding = dimensionResource(id = R.dimen.tab_tray_list_bottom_padding)
-    val state = rememberLazyListState()
 
-    LazyColumn(
-        modifier = Modifier.fillMaxSize(),
-        state = state,
-    ) {
-        items(
-            items = tabs,
-            key = { tab -> tab.id },
-        ) { tab ->
-            TabListItem(
-                tab = tab,
-                onCloseClick = onTabClose,
-                onMediaClick = onTabMediaClick,
-                onClick = onTabClick,
-                onLongClick = onTabLongClick,
-            )
-        }
+    if (displayTabsInGrid) {
```

Comment:
It still kinda feels to me like these separate cases should be separate file-private Composables. If it's a PITA though I think we can just leave as is

Reply:
It's not biggie to separate them back out.

Are you thinking like:

```
 fun TabLayout() {
    if (displayTabsInGrid) {
       TabGrid()
    } else {
        TabList()
    }
}

private fun TabList() 

private fun TabGrid() 

```

Still keeping the root conditional node with private leafs for the actual layouts?

Reply:
Yeah, that's what I'd been thinking!